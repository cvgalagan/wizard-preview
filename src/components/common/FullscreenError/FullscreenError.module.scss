@use "../../../styles/common";

$title-spacing: common.$font-size-base * 1.28;

.fullscreenError {
  &__content {
    display: flex;
    flex-direction: column;
    text-align: center;
  }
  &__title {
    user-select: none;
    margin-bottom: $title-spacing;
  }
  &__image {
    margin-top: 10px;
    user-select: none;
  }
  &__message {}
}