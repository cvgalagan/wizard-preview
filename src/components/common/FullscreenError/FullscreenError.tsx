import React from "react"
import styles from "./FullscreenError.module.scss"
import PublicPage from "../PublicPage/PublicPage"
import logo from "../../../logo.svg"

export interface FullscreenErrorProps {
    title?: string
    message?: string
}

const FullscreenError: React.FC<FullscreenErrorProps> = props => {
    const { title, message } = props
    return (
        <PublicPage
            hideLanguageSelector={true}
            hidePrivacyPolicy={true}
            hideTermOfUse={true}
            className={styles.fullscreenError}
            contentInCenter
        >
            <div className={styles.fullscreenError__content}>
                <h1 className={styles.fullscreenError__title}>{title}</h1>
                {message && <p className={styles.fullscreenError__message}>{message}</p>}
                <div className={styles.fullscreenError__image}>
                    <img src={logo} className="App-logo" alt="logo" />
                </div>
            </div>
        </PublicPage>
    )
}

export default FullscreenError
