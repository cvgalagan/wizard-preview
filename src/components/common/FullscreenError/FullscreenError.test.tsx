import React from "react"
import FullscreenError from "./FullscreenError"
import { shallow } from "enzyme"
import { mockedTranslation } from "../../../utility/tests/mockedObjects"

jest.mock("react-i18next", () => ({
    useTranslation: () => mockedTranslation
}))

const props = {
    title: "Test title",
    message: "Test message"
}

describe("FullscreenError", () => {
    it("should render", () => {
        const wrapper = shallow(<FullscreenError {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
    it("should render with title and message", () => {
        const wrapper = shallow(<FullscreenError {...props} />)
        expect(wrapper.find(".fullscreenError__title").text()).toBe(props.title)
        expect(wrapper.find(".fullscreenError__message").text()).toBe(props.message)
    })
    it("should render without title and message", () => {
        shallow(<FullscreenError />)
    })
})
