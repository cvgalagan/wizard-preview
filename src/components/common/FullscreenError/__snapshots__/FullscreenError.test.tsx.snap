// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullscreenError should render 1`] = `
<PublicPage
  className="fullscreenError"
  contentInCenter={true}
  hideLanguageSelector={true}
  hideLogo={false}
  hidePrivacyPolicy={true}
  hideTermOfUse={true}
>
  <div
    className="fullscreenError__content"
  >
    <h1
      className="fullscreenError__title"
    >
      Test title
    </h1>
    <p
      className="fullscreenError__message"
    >
      Test message
    </p>
    <div
      className="fullscreenError__image"
    >
      <img
        alt=""
        src="ctError.png"
      />
    </div>
  </div>
</PublicPage>
`;
